programs/client/Suggest.cpp

```diff
@@ -85,58 +184,81 @@ Suggest::Suggest()
-        " UNION ALL "
-        "SELECT metric FROM system.asynchronous_metrics"
-        " UNION ALL "
-        "SELECT metric FROM system.metrics"
-        " UNION ALL "
-        "SELECT macro FROM system.macros"
-        " UNION ALL "
-        "SELECT policy_name FROM system.storage_policies"
-        " UNION ALL "
-        "SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate";
+    std::vector<std::string> completion =
+    {
+        { "SELECT name FROM system.functions" },
+        { "SELECT name FROM system.table_engines" },
+        { "SELECT name FROM system.formats" },
+        { "SELECT name FROM system.table_functions" },
+        { "SELECT name FROM system.data_type_families" },
+        { "SELECT name FROM system.merge_tree_settings" },
+        { "SELECT name FROM system.settings" },
+        { "SELECT cluster FROM system.clusters" },
+        { "SELECT name FROM system.errors" },
+        { "SELECT event FROM system.events" },
+        { "SELECT metric FROM system.asynchronous_metrics" },
+        { "SELECT metric FROM system.metrics" },
+        { "SELECT macro FROM system.macros" },
+        { "SELECT policy_name FROM system.storage_policies" },
+        { "SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate" },
+    };
 
     /// The user may disable loading of databases, tables, columns by setting suggestion_limit to zero.
     if (suggestion_limit > 0)
     {
         String limit_str = toString(suggestion_limit);
-        query <<
-            " UNION ALL "
-            "SELECT name FROM system.databases LIMIT " << limit_str
-            << " UNION ALL "
-            "SELECT DISTINCT name FROM system.tables LIMIT " << limit_str
-            << " UNION ALL "
-            "SELECT DISTINCT name FROM system.dictionaries LIMIT " << limit_str
-            << " UNION ALL "
-            "SELECT DISTINCT name FROM system.columns LIMIT " << limit_str;
+        std::vector<std::string> extended_completion =
+        {
+            { "SELECT name FROM system.databases LIMIT " + limit_str },
+            { "SELECT DISTINCT name FROM system.tables LIMIT " + limit_str },
+            { "SELECT DISTINCT name FROM system.dictionaries LIMIT " + limit_str },
+            { "SELECT DISTINCT name FROM system.columns LIMIT " + limit_str },
+        };
+        std::copy(extended_completion.begin(), extended_completion.end(), std::back_inserter(completion));
+    }
+
+    /// EXISTS for each table maybe heavy, especially in debug build,
+    /// by default do it only if the version of the server does not match with the version of the client.
+    if (force_check)
+    {
+        if (force_check.value())
+        {
+            std::erase_if(completion, [&](const std::string & query)
+            {
+                return !queryTableExists(connection, timeouts, query);
```

Comment:
It may cause too many queries for each table.

Reply:
Agree, let's prefer your patch.

Reply:
Just one more `show tables in system` query is ok.